### ISSUE TITLE
All operations hang

### ISSUE THREAD

**Issue by viper1s** [Nexus-Mods/Nexus-Mod-Manager#681]
When performing any operation , NMM hangs

**Describe the bug** 
Moving 1 esp up one in the load order hangs NMM for about 3 minutes

Moving 5 and NMM stops responding . When closing the NMM with task manager the NMM has encountered a bug message appears

Uninstalling a mod , no matter the size and it takes 5 minutes each

Installing a mod , no matter the size , takes a really long time
 

**Environment**
- NMM version: _65.11_
- Mod(s) used: _<link to mod(s) on Nexus>_
Any and all

_Please remember to attach any trace logs you get in connection to this problem. Without these it will often be very hard to track down the bug._

**To Reproduce**
Moving one mod in the load order
Did not happen in the previous versions

**Additional information** 
_Anything that could help track down this problem, other software used, screenshots, etc._

**Comment by sphynxSR2** [comment]
I second this as I'm having the exact same issue. Glad that someone else reported it.

**Comment by squid-box** [comment]
The only code change related to this from 0.65.10 to 0.65.11 is [this line](https://github.com/Nexus-Mods/Nexus-Mod-Manager/commit/65ae5b591ecebcd3910183096f37177e2d481c7c#diff-340c9c0cf817978ff707f94c6f5c1e1aR319), which is just one more string comparison - it would not account for the crazy time increase you're describing.

I'm also not able to reproduce it on my machine, but I only have 57 plugins active. Is there anything special about your setup?

**Comment by sphynxSR2** [comment]
I have tons of mods and 250 .esp files but I can confirm it's the 0.65.11 that's having this problem. I reverted to 0.65.10 and it works perfectly. I don't know what do you mean by "anything special about your setup". I have a very good computer tbh.

**Comment by viper1s** [comment]
Nothing special about system. This issue only started with the 65.11 version. 
I do have about 100 + mods loaded. I would not think that the number mattered , as I had basically the same mod count before . When a mod gets updated , I download and uninstall the previous version before installing the updated. That is how I noticed.
One thing more I have noticed. On uninstall  , I am getting a higher "this mod failed to uninstall" errors and the red circle (disabled) shows. Even activating and then trying to uninstall again errors out. If I shut down mod manager and bring it right back up , the mod uninstalls correctly.

**Comment by Wishkahkid** [comment]
I'm having the same issue. I also had to revert back to 0.65.10, which doesn't have that problem.

**Comment by viper1s** [comment]
One other thing. When installing a mod there is no (or very trace amount) of CPU usage  , 384 mb of RAM , and no disk activity
Actually , there is just a minimal mount of CPU usage when moving a mod in the load order
Almost as if there is absolutely nothing going on , other than the straight 384 mb of memory in use during any operation.

**Comment by viper1s** [comment]
Got it. When installing a mod in which no other mod has a dependency on , the installation process is near normal. It seems the tremendous slow down is moving a mod either manually or automatically when installing
Something to look at.

**Comment by squid-box** [comment]
viper1s: Do you have any examples of mods / mod combinations where it happens? Would make debugging this much easier.

**Comment by viper1s** [comment]
Say one is installing a mod that a mod already in the load order is depending on the installing mod loading first. Both mods will have to shift in load order (most likely to the very bottom) due to the auto sort function of NMM . 

Or , it can be just moving one mod in the load order.
I found this out sort of by accident as I decided to install a mod for FO4 that I had never used. As the load order did not have to shift , it installed as efficiently as any mod in the previous versions of NMM

In other words , any time a load order is changed in any fashion , it takes a long time.
I hope this helps.
Yet , I do not see how that one line code changed anything in the fashion of load orders.
It is almost like the load order change goes into an almost infinite loop somewhere.

**Comment by viper1s** [comment]
After endless hours of troubleshooting and bottles of Tylenol , the only thing I have is the issue may be NMM's auto-sort function . If I remember correctly , there used to be a way to disable auto-sort by right clicking on the menu bar of the plugin section . I am unable to find that option now.
If that has not been tried , that may be a good next step

**Comment by viper1s** [comment]
UPDATE - 
Windows 10 patch from yesterday "seemed" to resolve the issue.
Go figure.
The issue of the very slow install appears resolved
The issue with the very slow plugin move in load order is not

**Comment by ghost** [comment]
I've just recorded [a video of the issue](https://mega.nz/#!zAp2xY7J!cmEFzbb10K0Gc2teKib0TgjkaNixNx-PpMXvVl_Rbeg). Reading these reports made me think it was an 0.65.11 issue, but I can reproduce this in 0.65.10 as well.

This issue has caused my NMM to improperly install a mod and leave leftover files after installation.

**Comment by viper1s** [comment]
I just got around to trying NMM on my other system
AMD Ryzen 5 2600x
16Gbs of DDR4 3000
Nvidia 1080Ti
Windows 10 1809
Never had NMM installed before.
Same issue except that installation of mods went a bit faster.

**Comment by ghost** [comment]
I've tested again with the latest 0.70.x versions and I can no longer reproduce this bug. Please consider closing this issue.

**Comment by sphynxSR2** [comment]
Nope. 0,70,7 here and still, same problem.

**Comment by sphynxSR2** [comment]
0.70.8, and having the same issue. I'm getting pretty frustrated as I don't get how the old version was working perfectly but the newer ones don't.

**Comment by squid-box** [comment]
I understand you're frustrated, but it's very hard to do anything about this since neither of the developers are able to reproduce the problem.

Could you force a trace log creation by following the instructions on this wiki page:

https://github.com/Nexus-Mods/Nexus-Mod-Manager/wiki/Bug-reporting

And then reproduce the error, and upload the created trace log here? It might show something about what's going on in the background.

**Comment by viper1s** [comment]
This problem is frustrating. However , I am not for sure this is a NMM issue. I think the issue may lie in Windows versions and/or drivers (at least partially). I have notice changes in the efficiency of NMM change back and forth after these updates . It was still slow , yet the changes were readily noticeable. 
As a key , "Bodyslide" for FO4 and SkyrimSE once was very quick . Now if I try and run this , the system locks up with drive light activity (which is a lot like this NMM issue) . I have not been patient enough after 20 minutes to see if this application will ever finish.

**Comment by sphynxSR2** [comment]
Ok, done. I opened NMM in trace mode and moved a plugin in the load order and then moved it back.  The log file looks like a mess to me and I barely understand a single thing but I think that the last lines are what you're looking for.

"Preparing to write load order to file" and so on...
[TraceLog20190622163059.txt](https://github.com/Nexus-Mods/Nexus-Mod-Manager/files/3316929/TraceLog20190622163059.txt)

**Comment by viper1s** [comment]
One more bit of strangeness with the "slow" moving of an object in the load order in NMM
I cannot move any icon on my Windows desktop while NMM is working on it.
It highlights , but I cannot move it. 
That is reproducible , and happens every single time

**Comment by viper1s** [comment]
After taking some time and getting to the RCA
The answer is -
Virtual Install
And , I noticed that in the period of time I am moving a file in the load order , I cannot move a file , folder , etc. on my Windows desktop. NMM actually locks it

**Comment by jbostrus** [comment]
Could anyone who was previously experiencing this issue please re-test with 0.70.10. I believe I resolved it. I'm pretty sure the described issue is the same as the other issue I requested be closed from this latest update.

**Comment by viper1s** [comment]
Just moved a few esps in both FO4 and SkyrimSE . Went smooth
Will continue testing when I have a chance.

**Comment by Wishkahkid** [comment]
I just tried it with FO4 and it works great! Thank you very much for fixing it, much appreciated.

**Comment by viper1s** [comment]
Ok , was able to get in more testing
What I have found
1) Smaller mods no longer take 4 minutes or so to install
2) I was able to move multiple esps in the load order of both SkyrimSE and FO4 . Was almost immediate .
Thanks for your diligence in fixing this most annoying issue.

**Comment by vonpire** [comment]
This isn't Fixed at all!

This bug is still present and will make me change NMM for good if isn't addressed. From my analysis this is due to mods that have dependencies, when installing or uninstalling it takes sometimes more than 30 mins, because of the autosort thing!! Mods without any dependency install/unistall "normal" !!!!! Please make a option to disable autosort for good! People use loot anyway! Skyrim SE! This bug exhists since 2019! Cmon!
